Component.vue

```diff
@@ -0,0 +1,29 @@
+<script setup>
+import { ref } from "vue";
+import axios from "axios";
+const pageTitle = "Data Fetching Example";
+const isLoading = ref(false);
+const data = ref([]);
+const fetchData = async () => {
+  try {
+    isLoading.value = true;
+    const response = await axios.get("https://api.example.com/data");
+    data = response.data;
+  } catch (error) {
+    console.error("Error fetching data:", error);
+  } finally {
+    isLoading.value = false;
+  }
+};
```

Comment:
`axios.get` returns a promise, so the "try/catch/finally" branches aren't necessary
```suggestion
const fetchData = () => {
  isLoading.value = true;
  axios.get("https://api.example.com/data").then((response) => {
      data.value = response.data;
      // no need to set `isLoading.value to false`, as the `.finally` branch will take care of it
    }).catch((errors) => {
      console.error("Error fetching data:", errors);
      // no need to set `isLoading.value to false`, as the `.finally` branch will take care of it
    }).finally(() => {
      isLoading.value = false;
    });
};
```